libcontainer/process_linux.go

```diff
@@ -487,6 +590,19 @@ func (p *initProcess) start() (retErr error) {
 		return fmt.Errorf("error waiting for our first child to exit: %w", err)
 	}
 
+	// Spin up a goroutine to handle remapping mount requests by runc init.
+	// There is no point doing this for rootless containers because they spawn
+	// the idmap processes, nor can they configure MOUNT_ATTR_IDMAP.
```

Comment:
In pure rootless, it's not possible because both `open_tree(OPEN_TREE_CLONE)` and `mount_setattr` require `may_mount()` (`ns_capable(current->nsproxy->mnt_ns->user_ns, CAP_SYS_ADMIN)`) which we don't have with rootless containers. We could pass plain `open(2)` paths for bind-mounts but there's no point because in that case there is no permission issue with bind-mounts (you cannot `setgroups(2)` in rootless and you never change kuid/kgid).